corehq/apps/app_manager/views/apps.py

```diff
@@ -902,6 +902,27 @@ def rearrange(request, domain, app_id, key):
         return back_to_main(request, domain, app_id=app_id, module_id=module_id)
 
 
+@no_conflict_require_POST
+@require_can_edit_apps
+def reorder_child_modules(request, domain, app_id):
+    app = get_app(domain, app_id)
+    modules_by_parent_id = OrderedDict(
+        (m.unique_id, [m]) for m in app.modules if not m.root_module_id
+    )
+    orphaned_modules = []
+    for module in app.modules:
+        if module.root_module_id:
+            if module.root_module_id in modules_by_parent_id:
+                modules_by_parent_id[module.root_module_id].append(module)
+            else:
+                orphaned_modules.append(module)
+
+    app.modules = [m for modules in modules_by_parent_id.values() for m in modules]
```

Comment:
Nitpick: having `modules` twice inside the list comprehension is a little confusing.

Reply:
what would you suggest?  I need to join the lists that are the dict values.  Could also do:
```python
list(itertools.chain(*modules_by_parent_id.values()))
```

Reply:
I think my real complaint is I don't quite get what this function is doing unless I go look at other commits for context. We're reordering modules, but why? What order are they going to end up in? Plus I find nested list comprehensions confusing which isn't helping. (In general when reviewing I waffle over whether to criticize nested comprehensions as difficult to read or just accept they're something I need to get more comfortable with.)

I think if the overall purpose of the function were clearer I wouldn't get hung up on this line. How about a function-level comment? Or a more descriptive function name?

Reply:
That's fair - I'll add in a docstring.  Nested list comprehensions took me a bit to get, but what finally did it for me is just remembering that the for statements are in the same order as in a regular nested loop.  So these two are equivalent:
```python
players = [player for team in teams for player in team]

players = []
for team in teams:
    for player in team:
        players.add(player)
```
I do think reversing the order would make more sense, but eh, that's how python works, I guess.  Just a matter of getting used to it.  I mentally move `player` to the end.

Reply:
Sounds good. Thanks for the example - I shall study ;)